src/test/java/org/example/LoginTests.java

```diff
@@ -1,38 +1,46 @@
 package org.example;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.apache.logging.log4j.LogManager;
+import org.example.pages.HomePage;
+import org.example.pages.LoginPage;
 import org.junit.jupiter.api.DisplayName;
-import org.openqa.selenium.By;
-import org.openqa.selenium.WebElement;
-import org.openqa.selenium.support.events.EventFiringWebDriver;
 import org.testng.annotations.Test;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+
+import com.mongodb.diagnostics.logging.Logger;
 
 
+/**
+ * LoginTests is a class that extends the BaseTests class.
+ * This class contains tests related to the login functionality of the application.
+ * Each test method in this class should correspond to a specific aspect of the login functionality.
+ *
+ * @author Juan Ocampo
+ * @version 1.0
+ */
 @DisplayName("Example of a Test Class from Login Module")
 public class LoginTests extends BaseTests {
-
-    private EventFiringWebDriver driver;
-    private static final Logger logger = LogManager.getLogger(LoginTests.class);
-
-    @Test
-    @DisplayName("Example of a Login Test Method")
-    public void testLogin() {
-        // Find the username and password fields and the login button
-        WebElement usernameField = driver.findElement(By.id("username"));
-        WebElement passwordField = driver.findElement(By.id("password"));
-        WebElement loginButton = driver.findElement(By.id("loginButton"));
-
-        logger.info("Typing the credentials in LoginPage");
-        usernameField.sendKeys("your_username");
-        passwordField.sendKeys("your_password");
-
-        logger.info("Login Action Button");
-        loginButton.click();
-        logger.info("Clicked the login button.");
-
-        WebElement dashboardHeader = driver.findElement(By.id("dashboardHeader"));
-        assert dashboardHeader.isDisplayed() : "Login failed.";
-        logger.info("Login successful.");
-    }
-
-}
+	private static final Logger logger = (Logger) LogManager.getLogger(BaseTests.class);
+
+	/**
+	 * This method is a Base for Future LoginTest.
+	 * Check Successful Login Functionalities
+	 */
+	@Test
+	@DisplayName("Example of a Login Test Method")
+	public void testLogin() {
+
+		LoginPage loginPage = new LoginPage(driver, logger);
+		logger.info("Typing the credentials in LoginPage");
+		loginPage.fillUserNameField("Juan");
+		loginPage.fillPasswordField("MyPassword");
+
+		logger.info("Login Action Button");
+		HomePage homePage = loginPage.clickLoginButton();
```

Comment:
in my opinion it will be better to move the HomePage class initialization to the BaseTest (like LoginPage) and just call the homePage.checkPageIsDisplayed(). Coz in this case u will have to create the Home page object all the time when you run login (each test will produce the object creation for Home page, it will be leak of memory, we are don't want that, don't we?).

Reply:
You’re right, the HomePage should be initialized in all tests. The page was initialized in BaseTest.